### PR TITLE
List data source views

### PR DESCRIPTION
**Pull request by flvndvd**
## Description

This Pr refactors the existing endpoints to provide all data source views for a specific vault and the entire workspace, leveraging the permissions of the current user via the authenticator object. The `DataSourceViewType` has been augmented to include `DataSource` elements, which are necessary to display the data source views.

Endpoint details:
- `/w/[wId]/vaults/[vId]/data_source_views/` will display all data source views for a specific workspace.
- `/w/[wId]/vaults/[vId]` will show all managed data source views along with other data sources.
- `/w/[wId]/data_source_views` will list all data source views that current connected user has access to

## Risk

It might break the settings around vaults.

### CODE REVIEW

File: types/src/front/data_source_view.ts [hunk 1]
```diff
@@ -1,11 +1,22 @@
...
+import { DataSourceOrViewCategory } from "./api_handlers/public/vaults";
+import { ConnectorProvider, EditedByUser } from "./data_source";
 
 export interface DataSourceViewType {
+  category: DataSourceOrViewCategory;
+  connectorId: string | null;
+  connectorProvider: ConnectorProvider | null;
   createdAt: number;
+  // TODO(GROUPS_INFRA) Add support for edited by on data source view.
+  editedByUser?: EditedByUser | null;
```
Comment (Fraggle) [inline types/src/front/data_source_view.ts:11]: Shouldn't it be a list ?

Comment (Fraggle) [reply types/src/front/data_source_view.ts:11]: (or `lastEditedBy` ?)

**Review by Fraggle — APPROVED** [review]
Thanks !

File: front/pages/api/w/[wId]/vaults/[vId]/index.ts [hunk 2]
```diff
@@ -82,7 +79,9 @@ async function handler(
     case "GET":
       const all = [
         ...(await getDataSourceInfos(auth, vault)),
-        ...(await getDataSourceViewsInfo(auth, vault)),
+        ...(await DataSourceViewResource.listByVault(auth, vault))
```
[the remark this answers is not in the document]
Comment (flvndvd) [reply front/pages/api/w/[wId]/vaults/[vId]/index.ts:82]: Will be tackle in a follow up PR, tracked [here](https://github.com/dust-tt/dust/issues/6825).

File: types/src/front/data_source_view.ts [hunk 3]
```diff
@@ -1,11 +1,22 @@
 import { ModelId } from "../shared/model_id";
+import { DataSourceOrViewCategory } from "./api_handlers/public/vaults";
+import { ConnectorProvider, EditedByUser } from "./data_source";
 
 export interface DataSourceViewType {
+  category: DataSourceOrViewCategory;
+  connectorId: string | null;
```
[the remark this answers is not in the document]
Comment (spolu) [reply types/src/front/data_source_view.ts:7]: kk

**Review by spolu — APPROVED** [review]

**Pull request merged**